website/docs/gen-ai/inference/rayserve-ha.md

```diff
@@ -0,0 +1,208 @@
+
+With the above `RayService` configuration, we have enabled GCS fault tolerance for the Ray head Pod and the Ray cluster can recover from head Pod crashes without restarting all the Ray workers.
+
+Let's apply the above `RayService` configuration and check the behavior.
+
+```bash
+cd ../../gen-ai/inference/
+envsubst < mistral-7b-rayserve-inf2/ray-service-mistral-ft.yaml| kubectl apply -f -
+```
+
+The output should look like below
+
+```text
+namespace/mistral created
+secret/hf-token created
+rayservice.ray.io/mistral created
+ingress.networking.k8s.io/mistral created
+```
+
+Check the status of the Ray Pods in the cluster.
+
+```bash
+kubectl get po -n mistral
+```
+
+The Ray head and worker Pods should be in `Running` state as below.
+
+```text
+NAME                                         READY   STATUS    RESTARTS   AGE
+mistral-raycluster-rf6l9-head-hc8ch          2/2     Running   0          31m
+mistral-raycluster-rf6l9-worker-inf2-tdrs6   1/1     Running   0          31m
+```
+
+### Simulate Ray Head Pod Crash
+
+Simulate Ray head Pod crashing by deleting the Pod
+
+```bash
+kubectl -n mistral delete po mistral-raycluster-rf6l9-head-xxxxx
+pod "mistral-raycluster-rf6l9-head-xxxxx" deleted
+```
+
+We can see that the Ray worker Pod is still running when the Ray head Pod is terminated and auto-restarted. Please see the below screenshots from Lens IDE.
+
+![Head Pod Deletion](img/head-pod-deleted.png)
+
+![Worker Pod Uninterrupted](img/worker-pod-running.png)
+
+#### Test the Mistral AI Gradio App
+
+Let's also test our Gradio UI App to see whether it's able to answer questions while the Ray head Pod is deleted.
+
+Open the Gradio Mistral AI Chat application by pointing your browser to localhost:7860.
+
+Now repeat the Ray head Pod crash simulation by deleting the Ray head Pod as shown in the above steps.
+
+While the Ray head Pod is terminated and is recovering, submit questions into the Mistral AI Chat interface. We can see from below screenshots that the chat application is indeed able to serve traffic while the Ray head Pod is deleted and is recovering. This is because the RayServe service points to the Ray worker Pod which in this case is never restarted because of the GCS fault tolerance.
+
+![Gradio App Test HA](img/gradio-test-ft.png)
+
+![Gradio App Test 1](img/answer-1.png)
+
+![Gradio App Test Contd](img/answer-1-contd.png)
+
+For a complete guide on enabling end-to-end fault tolerance to your RayServe application, please refer to [Ray Guide](https://docs.ray.io/en/latest/serve/production-guide/fault-tolerance.html#add-end-to-end-fault-tolerance).
+
+## Cleanup
+
+Finally, we'll provide instructions for cleaning up and deprovisioning the resources when they are no longer needed.
+
+**Step1:** Delete Gradio App and mistral Inference deployment
+
+```bash
+cd gen-ai/inference/mistral-7b-rayserve-inf2
+kubectl delete -f gradio-ui.yaml
+kubectl delete -f ray-service-mistral.yaml
```

Comment:
should this be `ray-service-mistral-ft.yaml`

Reply:
yes, changed.